src/test/java/wolox/training/controllers/BookControllerTest.java

```diff
@@ -0,0 +1,185 @@
+package wolox.training.controllers;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.isA;
+import static org.mockito.BDDMockito.given;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import wolox.training.models.Book;
+import wolox.training.repositories.BookRepository;
+
+@WebMvcTest(BookController.class)
+class BookControllerTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @Autowired
+    private ObjectMapper mapper;
+
+    @MockBean
+    private BookRepository bookRepository;
+
+    @Test
+    void whenGetAllBooks_thenReturnJsonArray() throws Exception {
+        //Given
+        Book bookTest = new Book(null,"Mystery","Paolo", "image.png", "Sorcerer's Stone", "-", "Salamander","2000", 250,"123456789");
+        List<Book> allBooksTest = Arrays.asList(bookTest);
+
+        given(bookRepository.findAll()).willReturn(allBooksTest);
+
+        // When + Then
+        mvc.perform(MockMvcRequestBuilders.get("/api/books")
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(1)))
+                .andExpect(jsonPath("$[0].title", is(bookTest.getTitle())));
+
+        System.out.println("whenGetAllBooks Passed");
+    }
+
+    @Test
+    void whenGetABookByAuthor_thenReturnJsonArray() throws Exception {
+        //Given
+        Book bookTest = new Book(null,"Mystery","Paolo", "image.png", "Sorcerer's Stone", "-", "Salamander","2000", 250,"123456789");
+
+        given(bookRepository.findFirstByAuthor("Paolo")).willReturn(java.util.Optional.of(bookTest));
```

Comment:
The name should be a constant, this way you have a constant for all the tests, and if you have to modify, you modify in one place